contents/waallet.ts

```diff
@@ -11,4 +16,47 @@ export const config: PlasmoCSConfig = {
 console.log(
   "You may find that having is not so pleasing a thing as wanting. This is not logical, but it is often true."
 )
+
+/**
+ * Usage in the DApp:
+ *
+ * await (window as any).waallet.request({
+ *  method: "eth_someOpenWindowMethod",
+ *  params: { post: "post:window" },
+ *   });
```

Comment:
Would here become `window.postMessage({ method: "post:window", params: ... })`?

Reply:
I just tried to replace the code on the [DApp side Line52](https://github.com/oneleo/waallet-workspace/blob/f1a3e8805ddf22e94d96caef8b9342f821d6abdc/workspace/dapp/src/App.tsx#L52) with the following.
```typescript
window.postMessage({ post: "post:window" });
// OR
window.postMessage({ post: "post:window" }, window.location.origin);
```
The Extension's Contents did not receive the postMessage() request sent by the DApp...

In this scenario, it seems necessary to rewrite the Provider to enable the Extension to perform postMessage(), so that the Extension's Contents Script can receive it:
Method 1: Perform postMessage() within the specified eth_Method.
Method 2: Add a dedicated eth_Method for postMessage().
Method 3: Add the window.wallet.connect() method for postMessage(), similar to [this code Line21](https://github.com/oneleo/waallet-workspace/blob/f1a3e8805ddf22e94d96caef8b9342f821d6abdc/workspace/waallet/packages/provider/waallet.ts#L21-L23) snippet.

Reply:
The popup actually will not be triggered directly by dapp. It is a process within specific flow of our provider. For example, when dapp request an account from our provider, in order to keep user safe, our provider will show the popup in the middle of `eth_accounts`, only user accepts it then `eth_accounts` can return account address back to dapp.

Reply:
Can you update the comment to emphasize use `window.postMessage` in content script to trigger the callback?

Reply:
Got it! I think Cyan is referring to the idea I mentioned earlier as Method 2, indicating that a standalone method like windows.create is not necessary. The creation of the window should consistently include the authentication action with the user's agreement.

No problem! I'll think about how to express it more clearly!